papis/docmatcher.py

```diff
@@ -117,50 +132,57 @@ def return_if_match(
         if cls.parsed_search is None or cls.matcher is None:
             return match
 
+        tokens = []
         for p in cls.parsed_search:
-            match = (
-                doc if cls.matcher(doc, p.pattern, cls.match_format, p.doc_key)
-                else None)
-
-            if not match:
-                break
+            if not p.syntax:
+                tokens.append(
+                    "True"
+                    if cls.matcher(doc, p.pattern, cls.match_format, p.doc_key)
+                    else "False"
+                )
+            else:
+                tokens.append(p.string)
+
+        result = eval(" ".join(tokens))
+        match = doc if result else None
 
         return match
 
     @classmethod
     def set_search(cls, search: str) -> None:
         """Set the search for this instance of the matcher.
 
-            >>> DocMatcher.set_search('author:Hummel')
-            >>> DocMatcher.search
-            'author:Hummel'
+        >>> DocMatcher.set_search('author:Hummel')
+        >>> DocMatcher.search
+        'author:Hummel'
```

Comment:
Is this a formatting change or do the doctests not run if they're indented? I should really check that :worried:

Reply:
Not sure how this happened. Will revert.